### PR TITLE
Implement filtering on companies and company interests

### PR DESCRIPTION
**Pull request by ivarnakken**
# Description

Search on names and descriptions, and filter out inactive companies.

Will add company type filtering as well; but this is blocked by https://github.com/webkom/lego/pull/3702

# Result

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

<table>
    <tr>
        <td>Before</td>
        <td>After</td>
    </tr>
    <tr>
        <td>
            image
        </td>
        <td>
            image
        </td>
    </tr>
</table>

Search filtering has also been implemented for company interest! Very useful instead of having to go through all semesters manually ..

image

# Testing

- [ ] I have thoroughly tested my changes.

Tested that the search filtering works fine. There's some issues with the active / inactive checkbox ... it seems all inactive companies are still not retrieved by the selector?

---

Resolves ABA-1203
Resolves ABA-1205
Resolves ABA-1209

### CODE REVIEW

File: app/routes/company/components/CompaniesPage.tsx [hunk 1]
```diff
@@ -84,20 +89,48 @@ const CompanyList = ({ companies = [] }: CompanyListProps) => (
...
+    const searchMatch =
+      !query.search ||
+      company.name.toLowerCase().includes(query.search.toLowerCase()) ||
+      company.description?.toLowerCase().includes(query.search.toLowerCase());
+
+    const activeMatch = !query.showInactive || company.active !== false;
+
+    return searchMatch && activeMatch;
+  });
+};
```
Comment (norbye) [inline app/routes/company/components/CompaniesPage.tsx:111]: Isn't doing this in the frontend a bit of a bad idea? Considering that the companies page is paginated..

File: app/routes/company/components/CompaniesPage.tsx [hunk 2]
```diff
@@ -84,20 +89,48 @@ const CompanyList = ({ companies = [] }: CompanyListProps) => (
...
+
 const CompaniesPage = () => {
   const [expanded, setExpanded] = useState(false);
+  const { query, setQueryValue } = useQuery(companiesDefaultQuery);
 
-  const companies = useAppSelector(selectActiveCompanies<ListCompany>);
+  const companies = useAppSelector(selectAllCompanies<ListCompany>);
+  const inactiveCompanies = companies.filter((company) => !company.active);
+  console.log('inactiveCompanies', inactiveCompanies);
```
Comment (norbye) [inline app/routes/company/components/CompaniesPage.tsx:119]: 👀

**Review by norbye — CHANGES_REQUESTED** [review]

File: app/routes/company/components/CompaniesPage.tsx (hunk 1 above)
Comment (ivarnakken) [reply app/routes/company/components/CompaniesPage.tsx:111]: Yes indeed - I guess this is more like a mvp prototype .. I'll get around and see what I can do

File: app/routes/company/components/CompaniesPage.tsx (hunk 2 above)
Comment (ivarnakken) [reply app/routes/company/components/CompaniesPage.tsx:119]: ooooooooooppsiiiiiiii

File: app/routes/company/components/CompaniesPage.tsx (hunk 1 above)
Comment (ivarnakken) [reply app/routes/company/components/CompaniesPage.tsx]: Implemented! https://github.com/webkom/lego/pull/3706

File: app/routes/company/components/CompaniesPage.tsx [hunk 3]
```diff
@@ -84,31 +89,89 @@ const CompanyList = ({ companies = [] }: CompanyListProps) => (
...
+
 const CompaniesPage = () => {
   const [expanded, setExpanded] = useState(false);
+  const { query, setQueryValue } = useQuery(companiesDefaultQuery);
+
+  const unfilteredCompanies = useAppSelector(selectAllCompanies<ListCompany>);
+  const companies = useMemo(
+    () => filterCompanies(unfilteredCompanies, query),
+    [unfilteredCompanies, query],
+  );
```
Comment (norbye) [inline app/routes/company/components/CompaniesPage.tsx:122]: Delete these please, makes no sense to filter both in the frontend and in the backend

**Review by norbye — CHANGES_REQUESTED** [review]

File: app/routes/company/components/CompaniesPage.tsx [hunk 4]
```diff
@@ -188,36 +188,39 @@ const CompaniesPage = () => {
...
-          <Button
-            flat
-            className={styles.readMore}
-            onPress={() => setExpanded(false)}
-          >
-            Vis mindre
-          </Button>
-        </div>
+        {expanded && (
+          <div className={!expanded ? utilities.hiddenOnMobile : undefined}>
```
Comment (norbye) [inline app/routes/company/components/CompaniesPage.tsx:192]:
```suggestion
          <div className={utilities.hiddenOnMobile}>
```
No need for ternary after `expanded && `

Comment (ivarnakken) [reply app/routes/company/components/CompaniesPage.tsx]: nice catch!

**Pull request merged**